ocaml/xapi/xapi_xenops.ml

```diff
@@ -1698,8 +1698,8 @@ module Xenopsd_metadata = struct
             info "xenops: VM.import_metadata %s" txt ;
             maybe_persist_md ~__context ~self txt ;
             let module Client = (val make_client queue_name : XENOPS) in
-            Client.VM.import_metadata_async dbg txt
-            |> Client.TASK.destroy_on_finish dbg
+            let (_ : Task.id) = Client.VM.import_metadata_async dbg txt in
```

Comment:
Patch is the wrong way around. The change was already part of the previous commit, and this reverses it.